src/main.cpp

```diff
@@ -1712,10 +1716,12 @@ void static InvalidChainFound(CBlockIndex* pindexNew)
         log(pindexNew->nChainWork.getdouble()) / log(2.0), DateTimeStrFormat("%Y-%m-%d %H:%M:%S",
                                                                pindexNew->GetBlockTime()));
 
-    const CBlockIndex* pChainTip = mapBlockIndex[chainActive.Tip()->GetBlockHash()];
+    const CBlockIndex* pChainTip = GetChainTip();
```

Comment:
I'm pretty sure that this method is always under `cs_main` lock. Instead of locking it recursively, would be better to add a `cs_main` lock held assertion at the beginning of the function.